src/createVariant.ts

```diff
@@ -1,25 +1,25 @@
-import {
-  BaseTheme,
-  Dimensions,
-  RestyleFunctionContainer,
-  ResponsiveValue,
-} from './types';
+import {BaseTheme, RestyleFunctionContainer, ResponsiveValue} from './types';
 import createRestyleFunction from './createRestyleFunction';
 import {all, AllProps} from './restyleFunctions';
 import composeRestyleFunctions from './composeRestyleFunctions';
 
 const allRestyleFunctions = composeRestyleFunctions(all);
 
-const createVariant = <Theme extends BaseTheme = BaseTheme>({
-  property = 'variant',
+const createVariant = <
+  Theme extends BaseTheme,
+  TProps extends VariantProps<Theme, K, P> = VariantProps<Theme, keyof Theme>,
+  P extends keyof TProps | 'variant' = 'variant',
```

Comment:
I think the P type is being set too strictly here, I'm not allowed to use anything other than `'variant'` as the prop name. 
![image](https://user-images.githubusercontent.com/688415/87530605-d89d5000-c690-11ea-9057-1e6ce239dd7f.png)

This should be able to accept any string.

Reply:
Thanks for the catch, have to make a small fix to allow other prop names, if the Prop types aren't defined, but it would still be good to enforce the type safety if the user does pass in props. Should work like this now:
![image](https://user-images.githubusercontent.com/7143583/87621746-b500f680-c6de-11ea-9f0f-aa48a100c38b.png)

Reply:
@JoelBesada Just pushed some changes giving createVariant some 🔥  type safety and inference. Try it out and let me know how it works. Stress test it with custom properties and PropTypes and themes, etc. It should be able to pick up any incorrect setup no matter how/where it's used. It works best when defined inline with createRestyleComponent/useRestyle, etc, but it still works when its not.